### PR TITLE
Make storage window a singleton

### PR DESCRIPTION
**Pull request by metalgearsloth**
It's how it works in ss13 but I expect mucho complaining.

:cl:
- tweak: Storage window is now a singleton and only one can be open at a time.

### CODE REVIEW

**Comment by Psychpsyo** [comment]
If this means that I can't open my backpack and tool belt at once then that'd be very annoying since I currently keep them open in the corner permanently while also interacting with other storages.

**Comment by deltanedas** [comment]
yes thats the current meta have all your storages open and never close them

**Comment by Tynobeard** [comment]
> yes thats the current meta have all your storages open and never close them

I already stopped doing this now that you can’t shrink container windows to be super slim.

**Comment by Equivocateur** [comment]
> If this means that I can't open my backpack and tool belt at once then that'd be very annoying since I currently keep them open in the corner permanently while also interacting with other storages.

That's the point. Not more modernized UI gamers putting all the storage in the black bars

**Comment by Psychpsyo** [comment]
> That's the point. No more modernized UI gamers putting all the storage in the black bars

I'm not sure what that accomplishes though apart from making inventory management more tedious.

**Comment by JoeHammad1844** [comment]
but why? just because it's in 13? whats wrong with keeping your backpack and belt open at the same time?

**Comment by metalgearsloth** [comment]
> but why? just because it's in 13? whats wrong with keeping your backpack and belt open at the same time?

People keeping 5 windows of heals open across backpack + belt + sub-storage in case they get into combat.

**Comment by JoeHammad1844** [comment]
> People keeping 5 windows of heals open across backpack + belt + sub-storage in case they get into combat.
what exactly do you see wrong with that?

**Comment by VektorZ1** [comment]
I wish there was a design document explaining why it's so cool and unusual to make inventory management less convenient

**Comment by metalgearsloth** [comment]
> > People keeping 5 windows of heals open across backpack + belt + sub-storage in case they get into combat.
> 
> what exactly do you see wrong with that?

I have a 4k screen which gives me an advantage over someone with 720p which is a bit shit, plus we are a remake.

**Comment by LordEclipse** [comment]
Aw man

**Comment by Visne** [comment]
I always keep my backpack open because it's convenient and I like knowing what I have with me, yet I avoid combat whenever I can. This is definitely not a feature that is only used by powergamers.

**Comment by JoeHammad1844** [comment]
> I have a 4k screen which gives me an advantage over someone with 720p which is a bit shit, plus we are a remake.

just because we are a remake doesnt mean we need to copy everything over 1:1, you're also a little inconsistent with that yourself as you (from memory) have denied my PR's to give the sec glasses sec hud (3PR's lmao)

**Comment by JoeHammad1844** [comment]
but that isn't really the main point

**Comment by JoeHammad1844** [comment]
having multiple containers open is a quality of life change and changing it makes the UX with the UI more confusing as you would have to scramble to hit the tiny X and then open the second container. it just makes everyone's experience worse

**Comment by metalgearsloth** [comment]
> just because we are a remake doesnt mean we need to copy everything over 1:1, you're also a little inconsistent with that yourself as you (from memory) have denied my PR's to give the sec glasses sec hud (3PR's lmao)

Because you re-open the same PR several times trying to sneak it through when several maintainers tell you no. I am telling you again don't post useless comments here.

**Comment by JoeHammad1844** [comment]
> Because you re-open the same PR several times trying to sneak it through when several maintainers tell you no. I am telling you again don't post useless comments here.

how is that a useless comment?

**Comment by LordEclipse** [comment]
Well how would it work, do you need to close the current window to open another one, or would opening a new window replace the current one and make it a new window? 

I know for sure this will make sec way more rough to play conveniently in general. I don't see why you wouldn't just stuff all your important stuff into one big bag and keep that one open all the time anyway removing any difference between the ability to  open mu!tiple containers and only being able to open one.
Also if you're worried about one person with a bigger screen having an advantage, that wasn't a problem when you could freely adjust the bag window size.

**Comment by metalgearsloth** [comment]
> don't think we should do this until storage UI is better, this wouldn't work nearly as well as it does in ss13

better how?

**Comment by metalgearsloth** [comment]
> Well how would it work, do you need to close the current window to open another one, or would opening a new window replace the current one and make it a new window?
> 
> I know for sure this will make sec way more rough to play conveniently in general. I don't see why you wouldn't just stuff all your important stuff into one big bag and keep that one open all the time anyway removing any difference between the ability to open mu!tiple containers and only being able to open one.
> 
> Also if you're worried about one person with a bigger screen having an advantage, that wasn't a problem when you could freely adjust the bag window size.

1. It's the one window so just gets adjusted for the new values.
2. Nested containers should be a trade-off and not something freely available.
3. Being able to make container windows 1px high is also undesirable behaviour.

This also opens the opportunity for stripping storage without instantly alerting a gamer who keeps all their storage windows open at all times.

> I always keep my backpack open because it's convenient and I like knowing what I have with me, yet I avoid combat whenever I can. This is definitely not a feature that is only used by powergamers.

This PR doesn't affect that.

**Comment by Visne** [comment]
> This PR doesn't affect that.
It does affect that, because now every time I open anything else I have to re-open my backpack. It's just annoying and nothing else, really.

**Comment by LordEclipse** [comment]
> 2. Nested containers should be a trade-off and not something freely available.

So like, boot knifes? (all of security) or coats? (all of science)

**Comment by Ilya246** [comment]
this also makes lower ping even stronger, even if slightly

**Comment by LordEclipse** [comment]
At the very least make it a cvar instead of hardcoded.

**Comment by nikthechampiongr** [comment]
I am not really sure if this is a good idea. Shuffling through your inventory is already a pain and the UI is quite clunky. Having to do it through only one window would only make this problem worse. The only other real conveniences is the hotkeys for the belt and backpack which allow you to take out the last item you put in there or put it back but this limits you to having a certain configuration for them. Having multiple windows open isn't about just being ready to fight, but just basic convenience.

**Comment by Equivocateur** [comment]
> > Because you re-open the same PR several times trying to sneak it through when several maintainers tell you no. I am telling you again don't post useless comments here.
> 
> how is that a useless comment?

You bringing up sloth has denied your sechud PR thrice after being told numerous times the word "no" is irrelevant to this PR.


> satchel new powergamer meta fast and low storage but stays open
backpack alternative meta fast and now high storage but does not stay open
duffelbag slow and high storage but stays open

* Items literally fall out of non-belt storage if container is open and you slip
* have to use the backpack or duffel in hand to access storage
* satchels 75 storage

Additionally can I throw in the fact that 131 stor duffels the ones that syndies have exclusively makes no sense and exists only because of hardsuits?

**Comment by mirrorcult** [comment]
> > don't think we should do this until storage UI is better, this wouldn't work nearly as well as it does in ss13
> 
> better how?

theres no reason to have window-based storage if the window is a singleton. if we're doing this then full commit to having it like ss13 where the storage just pops up in one predictable location thats easily accessible

**Comment by retequizzle** [comment]
not gonna lie i'm super skeptical over the reason for this and would like to propose a change to where this doesn't include utility belts/chest rigs/medical belts/sec belts, etc

if changes go through as is, sure you'd shaft a minority of players' minmax ability but it'd require sacrificing ease of access for (presumably) the majority. i know a LOT of players who play engi/med/sec keep their belt slot open for job related tasks and backpack open for general use, which is literally just two windows. i find it hard to believe that that one extra window is leading to that much of a skill gap between minmaxers and your average player. 

to that end, part of the issue is that players can hoard medical supplies so maybe we should limit it so that certain medical supplies can't be stored in non-medkit flagged containers. like, it makes sense that you wouldn't want to put sterilized medical equipment like medicated sutures into your backpack with your crowbar and wizard hat anyway, so implementing that would make more sense wouldn't it?

tl;dr - exempt belt/chestrig container from singleton so we don't impede on engi/med/sec gameplay more than necessary, shaft medkits/backpacks/other containers into singleton as proposed

**Comment by LordEclipse** [comment]
> tl;dr - exempt belt/chestrig container from singleton so we don't impede on engi/med/sec gameplay more than necessary, shaft medkits/backpacks/other containers into singleton as proposed

That's reasonable, theres no reason why you should be able to keep a medkit open while its in your backpack.

**Comment by metalgearsloth** [comment]
Continued at https://discord.com/channels/310555209753690112/770682801607278632/1153686215510470678

**Comment by EmoGarbage404** [comment]
worth reconsidering regarding new changes

**Comment by metalgearsloth** [comment]
Something else I was thinking is allowing multiple but only if they're nested, if you open an adjacent node on a higher level the ones below it close.

**Comment by Endecc** [comment]
>> [metalgearsloth in 21212:](https://github.com/space-wizards/space-station-14/pull/21212#issuecomment-1776997541) Limiting storage windows would favour lower resolution monitors I would think, no? At the moment higher-resolution monitors are favoured on master. But that's also a separate PR so I digress.
>
> [Me in 21212:](https://github.com/space-wizards/space-station-14/pull/21212#issuecomment-1777055217) If this is calling back to your PR https://github.com/space-wizards/space-station-14/pull/20335, then this still creates more UI fumbling, especially more so when encouraging box hoarding like this PR does. It's just adding tedium on top of tedium and doesn't make for any convenience or improvement. It's making things annoying for the sake of being annoying, which is not helpful for balance or usability.

Currently, I feel I shouldn't have to elaborate any further.

**Comment by metalgearsloth** [comment]
> Currently, I feel I shouldn't have to elaborate any further.

Leaving all storage windows open at once isn't desired behaviour so we welcome any solutions you present.

**Comment by QuietlyWhisper** [comment]
> > Currently, I feel I shouldn't have to elaborate any further.
> 
> Leaving all storage windows open at once isn't desired behaviour so we welcome any solutions you present.

I suggested adding dropdowns to containers
https://discord.com/channels/310555209753690112/1166733161573662810/1166765325270929460

**Comment by TemporalOroboros** [comment]
IMO this should allow one UI per top-level storage.
Eg: You can have one open for your backpack and belt. But if you want to open one for a box in your backpack that replaces the backpacks storage UI.

**Comment by TemporalOroboros** [comment]
Alternatively, just have one ui and require that the storage not be in a container to open it.

**Comment by Endecc** [comment]
> Leaving all storage windows open at once isn't desired behaviour so we welcome any solutions you present.

I suppose I _do_ have to elaborate.
Creating a system that relies heavily on nested storage menus [21212], and then making those menus more difficult to use [20334], is not user-friendly. It's introducing a speed-bump to slow things down, and then a toll gate to put things at a complete stop just in case it wasn't inconvenient enough.

Reducing the need of nested inventories, or in general the need of having access to multiple inventories at one time _(whether open at once, or needing to fetch them)_, is my core solution.

In my thinking, this could be done by either;
* Increasing the amount of "slots" in a Backpack, and reducing that of Boxes
* Having tiny items like paper or pens not count toward a full slot in two/three steps larger containers, like a backpack

This PR is not Slot-based storage so while I _would_ like to try to keep that direct feedback limited, it is heavily involved in making Singleton Storage windows **even more** unfriendly. And if it is here to stay, I find it important to include as a variable in my considerations.

**Comment by LankLTE** [comment]
> Creating a system that relies heavily on nested storage menus [21212], and then making those menus more difficult to use [20334], is not user-friendly. 

This is similarly how I feel. With this PR nested storage will still be optimal, it's just that it's also annoying to use, which isn't a great way to balance it out imo

**Comment by UbaserB** [comment]
I am interested in seeing how this would play out in the playerbase, to say the least.
